### PR TITLE
Feat: Add query docs

### PR DESCRIPTION
**Pull request by gewenyu99**
This file adds docs for the query class. This serves to display all available options, so users don't need to guess what query methods are available. This is a prerequisite for adding a more detailed explanation for the search parameter in various endpoints.

### CODE REVIEW

**Comment by gewenyu99** [comment]
Supported by this PR to homepage: https://github.com/appwrite/homepage/pull/47

**Comment by gewenyu99** [comment]
Screen Shot 2022-06-24 at 3 48 58 PM

File: app/views/docs/query.phtml [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Query } from "appwrite";
+const queryString = Query.equal(attribute, value);
+console.log(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Flutter</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Flutter SDK">
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:20]:
```suggestion
        <div class="ide" data-lang="dart" data-lang-label="Flutter SDK">
```

File: app/views/docs/query.phtml [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        <pre class="line-numbers"><code class="prism language-dart" data-prism>import 'package:appwrite/appwrite.dart';
+String queryString = Query.equal(attribute, value);
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:22]: Is Query automatically imported when importing the Appwrite package? In the web SDK, we need to import it separately.

File: app/views/docs/query.phtml [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+print(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Android</h3>
+        <div class="ide" data-lang="kotlin" data-lang-label="Android SDK">
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import 'package:appwrite/appwrite.dart';
+String queryString = Query.equal(attribute, value)
+print(queryString)</code></pre>
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:31]: This is the dart example, let's update it to use the Kotlin example

File: app/views/docs/query.phtml [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+<p>
+Returns results where the specified <b>attribute</b> is exactly equal to <b>value</b>.
+</p>
+<ul class="phases clear" data-ui-phases>
+    <li>
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Query } from "appwrite";
+const queryString = Query.equal(attribute, value);
+console.log(queryString);</code></pre>
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:15]: I see that we're printing the query string in each case, it seems unnecessary unless we show what the output looks like ... 

I think we can use two approaches here 

```js
import { Query } from "appwrite";

let attribute = "name";
let value = "Avengers";

const queryString = Query.equal(attribute, value);
console.log(queryString); // name=Avengers 
```

Comment (christyjacob4) [reply app/views/docs/query.phtml:15]: In all these examples , we should mention that `value` can either be a single value like `Avengers` or a list of values like `['Avengers', 'Titanic']`

Comment (christyjacob4) [reply app/views/docs/query.phtml:15]: Reference : https://github.com/appwrite/sdk-for-web/blob/master/src/sdk.ts#L3748-L3767

File: app/views/docs/query.phtml [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import 'package:appwrite/appwrite.dart';
+String queryString = Query.equal(attribute, value)
+print(queryString)</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>iOS</h3>
+        <div class="ide" data-lang="swift" data-lang-label="Apple SDK">
+            <pre class="line-numbers"><code class="prism language-swift" data-prism>import Appwrite
+let queryString = Query.equal(attribute, value)
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:38]: Please check if the query class is imported as part of the Appwrite package or needs an explicit import

File: app/views/docs/query.phtml [hunk 6]
```diff
@@ -0,0 +1,276 @@
+<p>
+Many Appwrite endpoints allows you to filter returned results through queries. Appwrite's SDKs include a <span class="tag">Query</span> class that helps construct query strings. This page documents all supported operations for Client SDKs. Server SDKs will behave similarly. 
+</p>
+
+<h2><a href="/docs/query#equal" id="equal">Equal</a></h2>
+<p>
+Returns results where the specified <b>attribute</b> is exactly equal to <b>value</b>.
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:7]: In all these examples , we should mention that value can either be a single value like `Avengers` or a list of values like `['Avengers', 'Titanic', 'Iron Man']`

Reference : https://github.com/appwrite/sdk-for-web/blob/master/src/sdk.ts#L3748-L3767

File: app/views/docs/query.phtml [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+            <pre class="line-numbers"><code class="prism language-swift" data-prism>import Appwrite
+let queryString = Query.equal(attribute, value)
+print(queryString)</code></pre>
+        </div>
+    </li>
+</ul>
+
+<h2><a href="/docs/query#not-equal" id="not-equal">Not Equal</a></h2>
+<p>
+Returns results where the specified <b>attribute</b> is not equal to <b>value</b>.
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:46]: In all these examples , we should mention that value can either be a single value like `Avengers` or a list of values like `['Avengers', 'Titanic', 'Iron Man']`

Reference : https://github.com/appwrite/sdk-for-web/blob/master/src/sdk.ts#L3748-L3767

File: app/views/docs/query.phtml [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Query } from "appwrite";
+const queryString = Query.notEqual(attribute, value);
+console.log(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Flutter</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Flutter SDK">
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:59]:
```suggestion
        <div class="ide" data-lang="dart" data-lang-label="Flutter SDK">
```

File: app/views/docs/query.phtml [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+        <pre class="line-numbers"><code class="prism language-dart" data-prism>import 'package:appwrite/appwrite.dart';
+String queryString = Query.notEqual(attribute, value);
+print(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Android</h3>
+        <div class="ide" data-lang="kotlin" data-lang-label="Android SDK">
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import 'package:appwrite/appwrite.dart';
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:68]: Replace with the Kotlin import

File: app/views/docs/query.phtml [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+            <pre class="line-numbers"><code class="prism language-swift" data-prism>import Appwrite
+let queryString = Query.notEqual(attribute, value)
+print(queryString)</code></pre>
+        </div>
+    </li>
+</ul>
+
+<h2><a href="/docs/query#less-than" id="less-than">Less Than</a></h2>
+<p>
+Returns results where the specified <b>attribute</b> is less than <b>value</b>. Can be applied to strings where comparison is based on alphabetical order.
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:85]: In all these examples , we should mention that value can either be a single value like `Avengers` or a list of values like `['Avengers', 'Titanic', 'Iron Man']`

Reference : https://github.com/appwrite/sdk-for-web/blob/master/src/sdk.ts#L3748-L3767

File: app/views/docs/query.phtml [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Query } from "appwrite";
+const queryString = Query.lesser(attribute, value);
+console.log(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Flutter</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Flutter SDK">
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:98]:
```suggestion
        <div class="ide" data-lang="dart" data-lang-label="Flutter SDK">
```

File: app/views/docs/query.phtml [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+        <pre class="line-numbers"><code class="prism language-dart" data-prism>import 'package:appwrite/appwrite.dart';
+String queryString = Query.lesser(attribute, value);
+print(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Android</h3>
+        <div class="ide" data-lang="kotlin" data-lang-label="Android SDK">
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import 'package:appwrite/appwrite.dart';
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:107]: Reminder to use the Kotlin import

File: app/views/docs/query.phtml [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+            <pre class="line-numbers"><code class="prism language-swift" data-prism>import Appwrite
+let queryString = Query.lesser(attribute, value)
+print(queryString)</code></pre>
+        </div>
+    </li>
+</ul>
+
+<h2><a href="/docs/query#less-than-equal" id="less-than-equal">Less Than or Equal To</a></h2>
+<p>
+Returns results where the specified <b>attribute</b> is less than or equal to <b>value</b>. Can be applied to strings where comparison is based on alphabetical order.
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:124]: In all these examples , we should mention that value can either be a single value like `Avengers` or a list of values like `['Avengers', 'Titanic', 'Iron Man']`

Reference : https://github.com/appwrite/sdk-for-web/blob/master/src/sdk.ts#L3748-L3767

File: app/views/docs/query.phtml (hunk 7 above)
Comment (christyjacob4) [reply app/views/docs/query.phtml:46]: You can use a numerical example to make it more clear

File: app/views/docs/query.phtml (hunk 10 above)
Comment (christyjacob4) [reply app/views/docs/query.phtml:85]: You can use a numerical example to make it more clear

File: app/views/docs/query.phtml (hunk 13 above)
Comment (christyjacob4) [reply app/views/docs/query.phtml:124]: You can use a numerical example to make it more clear

File: app/views/docs/query.phtml [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Query } from "appwrite";
+const queryString = Query.lesserEqual(attribute, value);
+console.log(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Flutter</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Flutter SDK">
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:137]:
```suggestion
        <div class="ide" data-lang="dart" data-lang-label="Flutter SDK">
```

File: app/views/docs/query.phtml [hunk 15]
```diff
@@ -0,0 +1,276 @@
...
+            <pre class="line-numbers"><code class="prism language-swift" data-prism>import Appwrite
+let queryString = Query.lesserEqual(attribute, value)
+print(queryString)</code></pre>
+        </div>
+    </li>
+</ul>
+
+<h2><a href="/docs/query#greater-than" id="greater-than">Greater Than</a></h2>
+<p>
+Returns results where the specified <b>attribute</b> is greater than <b>value</b>. Can be applied to strings where comparison is based on alphabetical order.
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:163]: In all these examples , we should mention that value can either be a single value like Avengers or a list of values like ['Avengers', 'Titanic', 'Iron Man']

Reference : https://github.com/appwrite/sdk-for-web/blob/master/src/sdk.ts#L3748-L3767

You can use a numerical example to make it more clear

File: app/views/docs/query.phtml [hunk 16]
```diff
@@ -0,0 +1,276 @@
...
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Query } from "appwrite";
+const queryString = Query.greaterEqual(attribute, value);
+console.log(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Flutter</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Flutter SDK">
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:176]:
```suggestion
        <div class="ide" data-lang="dart" data-lang-label="Flutter SDK">
```

File: app/views/docs/query.phtml [hunk 17]
```diff
@@ -0,0 +1,276 @@
...
+        <pre class="line-numbers"><code class="prism language-dart" data-prism>import 'package:appwrite/appwrite.dart';
+String queryString = Query.greaterEqual(attribute, value);
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:178]:
```suggestion
String queryString = Query. greater(attribute, value);
```

File: app/views/docs/query.phtml [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+print(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Android</h3>
+        <div class="ide" data-lang="kotlin" data-lang-label="Android SDK">
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import 'package:appwrite/appwrite.dart';
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:185]: Reminder to use the Kotlin import

File: app/views/docs/query.phtml [hunk 19]
```diff
@@ -0,0 +1,276 @@
...
+String queryString = Query.greaterEqual(attribute, value)
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:186]:
```suggestion
String queryString = Query. greater(attribute, value)
```

File: app/views/docs/query.phtml [hunk 20]
```diff
@@ -0,0 +1,276 @@
...
+            <pre class="line-numbers"><code class="prism language-swift" data-prism>import Appwrite
+let queryString = Query.greaterEqual(attribute, value)
+print(queryString)</code></pre>
+        </div>
+    </li>
+</ul>
+
+<h2><a href="/docs/query#greater-than" id="greater-than">Greater Than or Equal To</a></h2>
+<p>
+Returns results where the specified <b>attribute</b> is greater than or equal to <b>value</b>. Can be applied to strings where comparison is based on alphabetical order.
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:202]: In all these examples , we should mention that value can either be a single value like Avengers or a list of values like ['Avengers', 'Titanic', 'Iron Man']

Reference : https://github.com/appwrite/sdk-for-web/blob/master/src/sdk.ts#L3748-L3767

You can use a numerical example to make it more clear

File: app/views/docs/query.phtml [hunk 21]
```diff
@@ -0,0 +1,276 @@
...
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Query } from "appwrite";
+const queryString = Query.greaterEqual(attribute, value);
+console.log(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Flutter</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Flutter SDK">
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:215]:
```suggestion
        <div class="ide" data-lang="dart" data-lang-label="Flutter SDK">
```

File: app/views/docs/query.phtml [hunk 22]
```diff
@@ -0,0 +1,276 @@
...
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Query } from "appwrite";
+const queryString = Query.search(attribute, value);
+console.log(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Flutter</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Flutter SDK">
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:254]:
```suggestion
        <div class="ide" data-lang="dart" data-lang-label="Flutter SDK">
```

File: app/views/docs/query.phtml [hunk 23]
```diff
@@ -0,0 +1,276 @@
...
+        <pre class="line-numbers"><code class="prism language-dart" data-prism>import 'package:appwrite/appwrite.dart';
+String queryString = Query.search(attribute, value);
+print(queryString);</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Android</h3>
+        <div class="ide" data-lang="kotlin" data-lang-label="Android SDK">
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import 'package:appwrite/appwrite.dart';
```
Comment (christyjacob4) [inline app/views/docs/query.phtml:263]: Reminder to use the Kotlin import

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Pull request closed without merging**